[Azure OpenAI] Update billing dashboard panel

muthu-mps opened the pull request:
- Enhancement

## Proposed commit message

This PR changes the panel to display the cost over the time and updates the decimal places by 5 to show the actual costs which is lower.


## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- Install the Azure OpenAI integration
- View the billing dashboard panel that should display the costs between 0-1.



## Screenshots

<img width="1723" alt="azure_open_ai_billing" src="https://github.com/user-attachments/assets/66cf27ab-e23d-409f-818a-7e27d95e85a6" />

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/24580)
* Commit: 80a51c262e1be55283d8746bc8d2e32cb74de1c9

cc @muthu-mps

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24580","number":24580,"commit":"80a51c262e1be55283d8746bc8d2e32cb74de1c9"}],"number":24580}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/24586)
* Commit: 457dd9234db93757eaa8ea59022b7b81126d393a

### History
* :green_heart: [Build #24580](https://buildkite.com/elastic/integrations/builds/24580) succeeded 80a51c262e1be55283d8746bc8d2e32cb74de1c9

cc @muthu-mps

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24586","number":24586,"commit":"457dd9234db93757eaa8ea59022b7b81126d393a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24580","number":24580,"commit":"80a51c262e1be55283d8746bc8d2e32cb74de1c9"}],"number":24586}
buildkite-pr-comment-->

agithomas commented:
The first thing that comes to my mind is the need for having a currency value in the Y-axis. Is this doable?

agithomas commented:
Suggestion: While it is good to see the trend , showing the information of on-going month's usage cost / on-going day's usage cost would benefit the users, i think, which could be represented as a Single-stat / metric panel.

muthu-mps commented:
> The first thing that comes to my mind is the need for having a currency value in the Y-axis. Is this doable?

This is discussed earlier and currently we cannot add currency to the Y-axis. Even the Azure Billing integration dashboard doesn't include details related to the currency. When making changes to the Azure Billing integration dashboard I will check for some alternative ways to represent.

muthu-mps commented:
> Suggestion: While it is good to see the trend , showing the information of on-going month's usage cost / on-going day's usage cost would benefit the users, i think, which could be represented as a Single-stat / metric panel.

Do you mean adding one more metric panel to show the overall cost for the selected time? Can we take this in a separate ticket, WDYT?

agithomas commented on packages/azure_openai/kibana/dashboard/azure_openai-f5bbc591-5e6b-4af7-b6c5-b02065a06455.json:
```diff
@@ -340,7 +341,7 @@
                                     "shouldTruncate": true,
                                     "showSingleSeries": true
                                 },
-                                "preferredSeriesType": "bar_stacked",
```
Any specific reason why we went from bar_stacked to line? Generally i observe stacked bar chart to represent cost and usage details.

muthu-mps replied:
When we use the line chart with `line interpolation to smooth` this looks nice when compared to bar chart. Its also useful to represent to flow of cost.

daniela-elastic replied:
Pros and cons for both. 
Bar char: pros - you can see the daily cost. cons - you don't see how the fluctuations throughout the day
line chart: pros - you can see the daily fluctuations. cons - harder to convert this into a single daily number
I think either one is fine as long as we can have an additional widget which is for the total cost for the selected time period. Bonus points if we add an average daily cost for the selected time period.

muthu-mps replied:
cc: @daniela-elastic

daniela-elastic replied:
Thank Muthu. Agree that it makes sense to use bar charts if we have a 24 hour granularity. COuple of comments/questions: 
- For total costs we show 3 numbers after the decimal point. If it's a large number (larger than 0.009) we might want to show only 2 digits after the decimal point. However if we need to account for very small numbers (eg 0.009) then showing more numbers after the decimal point would make sense
- For total costs - what is the unit? Dollars? Yen?
- It's not clear what is the name of the second widget (the one to the right of  the "Total cost" widget). Is there a way to make it more obvious what it is?
- The bar chart widgets - they seem to have two parts - I think it's because of the vertical bar separating the months of March and April but it could also be easily interpreted as two widgets. Do we need to have this vertical bar?
- The daily usage widgets  - what units are those in? The first one says "EUR" but I would expect the unit to the shown somewhere close to the name of what this unit applies to (the y axis). ALso, is the other daily usage widget also in EUR?
- If my bill is in USD will I see the USD currency instead of the EUR currency on the dashboard?

daniela-elastic reviewed (commented):
added some suggestions. LMK what you think

daniela-elastic commented:
I also don't mind splitting the work in two tickets - the current one for the trend and another one for the totals

daniela-elastic commented:
we don't need a second panel for billing. we can add everything billing related to the same link panel

muthu-mps commented:
@agithomas , @daniela-elastic  - Based on the review comments, I have added metrics panel to show the overall trend. Also changed the line chart to bar chart.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/24944)
* Commit: a663b3356f384a0fc98af0068a9307aabd216e6f

### History
* :green_heart: [Build #24586](https://buildkite.com/elastic/integrations/builds/24586) succeeded 457dd9234db93757eaa8ea59022b7b81126d393a
* :green_heart: [Build #24580](https://buildkite.com/elastic/integrations/builds/24580) succeeded 80a51c262e1be55283d8746bc8d2e32cb74de1c9

cc @muthu-mps

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24944","number":24944,"commit":"a663b3356f384a0fc98af0068a9307aabd216e6f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24586","number":24586,"commit":"457dd9234db93757eaa8ea59022b7b81126d393a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24580","number":24580,"commit":"80a51c262e1be55283d8746bc8d2e32cb74de1c9"}],"number":24944}
buildkite-pr-comment-->

agithomas commented:
<img width="423" alt="image" src="https://github.com/user-attachments/assets/06cca092-8818-4b19-97bd-5dc642f08ed3" />

In here, we are not showing the total cost (in EUR), but the cost utilisation percentage.


Should one of the following be a better title? Or, show the cost instead of utilisation percentage.

- Model cost utilisation
- Percentage cost by Model
- Model-wise cost distribution.

muthu-mps commented:
Updated both the charts to display the currency code as part of the metric value.

### Currency code with metric chart:
I am able to get the currency code value but unlikely that the chart itself has some issues. It creates two layers even when there is only one value available. This seems to be an issue with Kibana itself.

<img width="1719" alt="Screenshot 2025-04-22 at 2 46 58 PM" src="https://github.com/user-attachments/assets/5ca6e24f-2a43-488b-b2eb-aefb4f1bf05c" />

Currently, Its not possible to show currency value in the total metrics chart.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/24965)
* Commit: 32b5cbc07a86b46a2ad5c2a82439e69f9c48ab56

### History
* :green_heart: [Build #24944](https://buildkite.com/elastic/integrations/builds/24944) succeeded a663b3356f384a0fc98af0068a9307aabd216e6f
* :green_heart: [Build #24586](https://buildkite.com/elastic/integrations/builds/24586) succeeded 457dd9234db93757eaa8ea59022b7b81126d393a
* :green_heart: [Build #24580](https://buildkite.com/elastic/integrations/builds/24580) succeeded 80a51c262e1be55283d8746bc8d2e32cb74de1c9

cc @muthu-mps

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24965","number":24965,"commit":"32b5cbc07a86b46a2ad5c2a82439e69f9c48ab56"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24944","number":24944,"commit":"a663b3356f384a0fc98af0068a9307aabd216e6f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24586","number":24586,"commit":"457dd9234db93757eaa8ea59022b7b81126d393a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24580","number":24580,"commit":"80a51c262e1be55283d8746bc8d2e32cb74de1c9"}],"number":24965}
buildkite-pr-comment-->

daniela-elastic commented on packages/azure_openai/img/azure_open_ai_billing.png:
Shouldn't the widget called "Model cost utilization" be named "Overall usage - by model" ? "Utilization" usually refers to usage as part of overall capacity (like CPU and memory utilization)

daniela-elastic reviewed (approved):
All  look good. Please update the widget based on my comment

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/25024)
* Commit: 3cdbbcb391bd9d6108a922df85ca40c75259b5e1

### History
* :green_heart: [Build #24965](https://buildkite.com/elastic/integrations/builds/24965) succeeded 32b5cbc07a86b46a2ad5c2a82439e69f9c48ab56
* :green_heart: [Build #24944](https://buildkite.com/elastic/integrations/builds/24944) succeeded a663b3356f384a0fc98af0068a9307aabd216e6f
* :green_heart: [Build #24586](https://buildkite.com/elastic/integrations/builds/24586) succeeded 457dd9234db93757eaa8ea59022b7b81126d393a
* :green_heart: [Build #24580](https://buildkite.com/elastic/integrations/builds/24580) succeeded 80a51c262e1be55283d8746bc8d2e32cb74de1c9

cc @muthu-mps

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/25024","number":25024,"commit":"3cdbbcb391bd9d6108a922df85ca40c75259b5e1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24965","number":24965,"commit":"32b5cbc07a86b46a2ad5c2a82439e69f9c48ab56"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24944","number":24944,"commit":"a663b3356f384a0fc98af0068a9307aabd216e6f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24586","number":24586,"commit":"457dd9234db93757eaa8ea59022b7b81126d393a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24580","number":24580,"commit":"80a51c262e1be55283d8746bc8d2e32cb74de1c9"}],"number":25024}
buildkite-pr-comment-->

agithomas commented:
@muthu-mps 

Can you update the screenshot, part of the package bundle, to reflect the most recent changes?

Suggestion: As we are doing more than changing the type of visualisation, it would be best to reflect that in the changelog entry as well.

agithomas reviewed (approved):
LGTM!

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/25031)
* Commit: 3f34a3269d82955d7895208ef30eb1fb0bd9160d

### History
* :green_heart: [Build #25024](https://buildkite.com/elastic/integrations/builds/25024) succeeded 3cdbbcb391bd9d6108a922df85ca40c75259b5e1
* :green_heart: [Build #24965](https://buildkite.com/elastic/integrations/builds/24965) succeeded 32b5cbc07a86b46a2ad5c2a82439e69f9c48ab56
* :green_heart: [Build #24944](https://buildkite.com/elastic/integrations/builds/24944) succeeded a663b3356f384a0fc98af0068a9307aabd216e6f
* :green_heart: [Build #24586](https://buildkite.com/elastic/integrations/builds/24586) succeeded 457dd9234db93757eaa8ea59022b7b81126d393a
* :green_heart: [Build #24580](https://buildkite.com/elastic/integrations/builds/24580) succeeded 80a51c262e1be55283d8746bc8d2e32cb74de1c9

cc @muthu-mps

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/25031","number":25031,"commit":"3f34a3269d82955d7895208ef30eb1fb0bd9160d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/25024","number":25024,"commit":"3cdbbcb391bd9d6108a922df85ca40c75259b5e1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24965","number":24965,"commit":"32b5cbc07a86b46a2ad5c2a82439e69f9c48ab56"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24944","number":24944,"commit":"a663b3356f384a0fc98af0068a9307aabd216e6f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24586","number":24586,"commit":"457dd9234db93757eaa8ea59022b7b81126d393a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24580","number":24580,"commit":"80a51c262e1be55283d8746bc8d2e32cb74de1c9"}],"number":25031}
buildkite-pr-comment-->
